src/php/Compiler/Domain/Analyzer/TypeAnalyzer/SpecialForm/MacroExpandSymbol.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Phel\Compiler\Domain\Analyzer\TypeAnalyzer\SpecialForm;
+
+use Phel\Compiler\Domain\Analyzer\Ast\MacroExpandNode;
+use Phel\Compiler\Domain\Analyzer\Environment\NodeEnvironmentInterface;
+use Phel\Compiler\Domain\Analyzer\Exceptions\AnalyzerException;
+use Phel\Compiler\Domain\Analyzer\TypeAnalyzer\WithAnalyzerTrait;
+use Phel\Lang\Collections\LinkedList\PersistentListInterface;
+use Phel\Lang\Symbol;
+
+use function count;
+
+final class MacroExpandSymbol implements SpecialFormAnalyzerInterface
+{
+    use WithAnalyzerTrait;
+
+    public function analyze(PersistentListInterface $list, NodeEnvironmentInterface $env): MacroExpandNode
+    {
+        if (!($list->get(0) instanceof Symbol
+            && $list->get(0)->getName() === Symbol::NAME_MACRO_EXPAND)
+        ) {
+            throw AnalyzerException::withLocation("This is not a 'macroexpand.", $list);
+        }
+
+        if (count($list) !== 2) {
+            throw AnalyzerException::withLocation("Exactly one argument is required for 'macroexpand", $list);
+        }
+
+        $form = $list->get(1);
+        if ($form instanceof PersistentListInterface
+            && $form->get(0) instanceof Symbol
+            && $form->get(0)->getName() === Symbol::NAME_QUOTE
+        ) {
+            $form = $form->rest()->first();
+        }
+
+        return new MacroExpandNode(
+            $env,
+            $this->analyzer->analyze($form, $env),
```

Comment:
@jenshaase, I think the missing part here is, how do we pass the actual macro content here, so it is treated as such (like a literal)? Is this what is missing, @smeghead ?

Reply:
Right now the output is this 
![Screenshot 2024-07-08 at 22 48 26](https://github.com/phel-lang/phel-lang/assets/5256287/83a03819-f609-4efe-bd7b-4aa3cbad3252)